rtmp-client/src/main/java/net/butterflytv/rtmp_client/RtmpClient.java

```diff
@@ -153,11 +378,16 @@ public int write(byte[] data) throws IOException {
      * @return number of bytes written
      * @throws IOException if connection is not opened or connection to server is lost
      */
-    public int write(byte[] data, int offset, int size) throws IOException {
-        return nativeWrite(data, offset, size, rtmpPointer);
+    public int write(byte[] data, int offset, int size) throws IOException, IllegalStateException {
+        rtmpError = RtmpIOException.RTMP_ERROR_NONE;
+        int ret = nativeWrite(data, offset, size, rtmpPointer);
+        if (ret < RtmpIOException.RTMP_ERROR_NONE) {
```

Comment:
nit - Use `!=` instead of `<`, so the reader of the code doesn't have to understand all error codes are smaller than `RTMP_ERROR_NONE` to understand the code.

Note: Probably needs removal of `RTMP_SUCCESS`.

Reply:
The function write returns the number of bytes written. Hence the `<` check in write and read.

Reply:
Oops; please ignore.